src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -192,6 +192,14 @@ type ExternalConnectivityConfig struct {
 	// If TLS is enabled then this subdomain will be requested
 	// as a subject alternative name.
 	Subdomain string `json:"subdomain,omitempty"`
+	// The preferred address type to be assigned to the external
+	// advertised addresses. The valid types are  External DNS,
+	// External IP, Internal DNS, Internal IP, Hostname.
+	// When the address of the preferred type is not found, the above
+	// types are tried in the same order. The default preferred
+	// address type is ExternalIP.
+	// This option only applies when Subdomain is empty.
+	PreferredAddressType string `json:"preferredAddressType,omitempty"`
```

Comment:
you have to be precise here otherwise it's hard to configure for customers. E.g. it's not External DNS but ExternalDNS. Also we need to make sure that ExternalDNS == subdomain you provided, right?

Reply:
Oh yes.. I had spaces in the comment for all types. Thanks.